Client/Frontend/Browser/LinkPreviewViewController.swift

```diff
@@ -9,33 +9,49 @@ import Shared
 
 class LinkPreviewViewController: UIViewController {
 
-  let url: URL
+  private let url: URL
+  private weak var parentTab: Tab?
+  private var currentTab: Tab?
+  private weak var browserController: BrowserViewController?
 
-  init(url: URL) {
+  init(url: URL, for tab: Tab, browserController: BrowserViewController) {
     self.url = url
+    self.parentTab = tab
+    self.browserController = browserController
     super.init(nibName: nil, bundle: nil)
   }
 
   @available(*, unavailable)
   required init?(coder aDecoder: NSCoder) { fatalError() }
 
   override func viewDidLoad() {
-    let configuration = WKWebViewConfiguration().then {
-      $0.setURLSchemeHandler(InternalSchemeHandler(), forURLScheme: InternalURL.scheme)
+    guard let parentTab = parentTab,
+          let tabWebView = parentTab.webView else {
+      return
     }
-
-    let wk = WKWebView(frame: view.frame, configuration: configuration)
-
-    let domain = Domain.getOrCreate(forUrl: url, persistent: !PrivateBrowsingManager.shared.isPrivateBrowsing)
-
-    BlocklistName.blocklists(forDomain: domain).on.forEach {
-      ContentBlockerHelper.ruleStore.lookUpContentRuleList(forIdentifier: $0.filename) { rule, _ in
-        guard let rule = rule else { return }
-        wk.configuration.userContentController.add(rule)
-      }
+    
+    currentTab = Tab(configuration: tabWebView.configuration,
+                     type: .private,
```

Comment:
per slack discussion we want to base this off tab's private mode?

Reply:
Fixed: https://github.com/brave/brave-ios/pull/5933/commits/34b1b748df12e8c23b40e38b9bd50f972a7b3036